instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py

```diff
@@ -341,6 +378,9 @@ def process_response(self, request, response):
             else:
                 activation.__exit__(None, None, None)
 
+        duration = max(round((default_timer() - request_start_time) * 1000), 0)
+        self._duration_histogram.record(duration, duration_attrs)
+        self._active_request_counter.add(-1, active_requests_count_attrs)
```

Comment:
Add `None` checks to not throw any exception in instrumentation path

Reply:
`.pop` raises `KeyError` if there is no key (which should never happen in instrumentation path) so you should pass a default `None` and if the value is None for some reason it throws TypeError when try to subtract None from float. Make sure the necessary checks are added and no exceptions are thrown in any case.

Reply:
@srikanthccv: KeyError will never come as we always set the keys in request.META in process_request() method.
I agree with your TypeError issue. I will add the check for the same

Reply:
No we have seen weird situations where keys always don't exist for some reason which is why we have None default and check for `span` and `activation` just above your code. Instrumentation must not raise any exception. We should be extra careful so please add the `None` default for `.pop`

Reply:
ok sure